x-pack/plugins/ui_actions_enhanced/public/actions/action_enhanced.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  Action,
+  ActionType,
+  UiActionsActionDefinition as ActionDefinition,
+} from '../../../../../src/plugins/ui_actions/public';
+import { LicenseType } from '../../../licensing/public';
+
+export interface ActionEnhanced<Context extends {} = {}, T = ActionType>
+  extends Action<Context, T> {
+  /**
+   * Minimal license
+   * if missing, then no license limitations
+   */
+  readonly minimalLicense?: LicenseType;
+
+  /**
+   * Does this action meet current licence?
+   */
+  isCompatibleLicence(): boolean;
+}
```

Comment:
We need to figure out with this enhancements pattern, how to handle state migrations. We know it's going to be a problem with the way Embeddables is set up. I touch on the problem here: https://gist.github.com/stacey-gammon/ad6e60637404e779bdaba8502e54ac03#going-over-the-corner-cases-with-a-specific-example

`UiActions` will call the enhanced migrators:

```
  state.enhancements = state.enhancements.forEach(key => 
    // What do we pass in for "key" here? How do we ensure uniqueness and avoid subtle bugs if an enhancer
    // fails to register a PersistableStateDefinition for `key`, and it accidentally matches some other
    // PersistableStateDefinition?
    persistableStatePlugin.get(key).migrate(
      state.enhancements[key],
      version)
    );
``` 

As you can see though, I don't have an answer yet for what to use for id (some combination of pluginId and the registry id? Like "uiAction-enhancedUiActions"?   Hopefully this will be fleshed out in  https://github.com/elastic/kibana/pull/67931.

cc @ppisljar